src/sagemaker_training/smdataparallel.py

```diff
@@ -432,3 +464,35 @@ def _parse_custom_mpi_options(custom_mpi_options):
     parser.add_argument("--NCCL_DEBUG", default="INFO", type=str)
 
     return parser.parse_known_args(custom_mpi_options.split())
+
+
+def _validate_smddp_coll_libs_present():
+    """Check if SMDDP collective communication libraries are present in the
+    container and have executable permissions. These libraries are installed in
+    SageMaker DLCs via an smddp-installer-<versions>.run file in the
+    deep-learning-containers package.
+
+    libsmddp.so is preloaded for training when SMDDP Collectives are used.
+    The other libraries libsmddpcoll.so, liboutofband.so and libgloo.so are
+    essential for successful training. Thus, if the libraries are absent,
+    we want to avoid training failure by disabling SMDDP Collectives.
+    """
+    smddp_lib_paths = [
```

Comment:
you can remove this smddp_lib_paths since we defined at top.

Reply:
My bad, fixed.